packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -334,17 +440,49 @@ export class LitDevTutorial extends LitElement {
   }
 
   protected renderFooter() {
-    return html`<div id="tutorialFooter">
-      <litdev-icon-button @click=${this._onClickSolve}>
-        ${solveIcon} Solve
-      </litdev-icon-button>
+    const snackbarLabel = `The code has ${
+      this._checkPassed ? 'passed' : 'failed'
+    } the checks${
+      this._validationMessage ? ` – ${this._validationMessage}` : '.'
+    }`;
 
-      <litdev-icon-button @click=${this._onClickReset}>
-        ${resetIcon} Reset
-      </litdev-icon-button>
+    return html`<div id="tutorialFooter">
+        <litdev-icon-button @click=${this._onClickSolve}>
+          ${solveIcon} Solve
+        </litdev-icon-button>
+
+        ${this._manifestTask.render({
+          complete: (manifest) =>
+            !manifest.steps[this._idx].checkable
+              ? nothing
+              : html` <litdev-icon-button
+                  class="checkButton ${this._isChecking ? 'checking' : ''}"
+                  @click=${this._onClickCheck}
+                  ?disabled=${this._isChecking || this._requestSolvedCode}
+                >
+                  ${this._renderCheckIcon()} Check
+                </litdev-icon-button>`,
+        })}
+
+        <litdev-icon-button @click=${this._onClickReset}>
+          ${resetIcon} Reset
+        </litdev-icon-button>
+
+        <span id="nextStep"> ${this._renderNextStepStatus()} </span>
+      </div>
+      <mwc-snackbar .labelText=${snackbarLabel}></mwc-snackbar>`;
+  }
 
-      <span id="nextStep"> ${this._renderNextStepStatus()} </span>
-    </div>`;
+  private _renderCheckIcon() {
+    if (this._checkIndeterminate) {
+      return checkCodeIcon;
+    } else if (this._isChecking) {
+      return loopIcon;
+    } else if (this._checkPassed) {
+      return checkPassedIcon;
+    } else {
+      return checkFailedIcon;
```

Comment:
Complexity reducing idea.

With three independent state variables, it creates 8 combinations of booleans to handle. I think the state can be simplified into a single `@state` with multiple enum options or intersection type strings, i.e. `'check-indeterminate' | 'is-checking' ...` modeling a state machine pattern with transitions between states. This will reduce the complexity of needing to change multiple variables for one state change. For example right now `isChecking` and `checkPassed` can both be true but I think it's redundant.

It seems like the states are:

 - `checkIndeterminate` for showing the `checkCodeIcon` and button being enabled.
 - `isChecking` to prevent multiple concurrent checks.
 - `check-success` for success.
 - `check-failure` for failure.

^ Each of the above states are exclusive so can be stored on one `@state`.
What do you think?